reconcile/utils/oc.py

```diff
@@ -1666,6 +1666,20 @@ def init_oc_client(self, cluster_info, privileged: bool):
                 )
                 return
 
+            token = token_secret[automation_token["field"]]
+            known_server_url = token_secret["server"]
+
+            if server_url != known_server_url:
+                self.set_oc(
+                    cluster,
+                    OCLogMsg(
+                        log_level=logging.ERROR,
+                        message=f"[{cluster}] server URL mismatch",
+                    ),
+                    privileged,
+                )
+                return
```

Comment:
This behaves now different from new `OCMap`. Should we also raise an exception here to stay consistent, i.e., fail hard? Alternatively, I will open a follow-up PR to adjust `OCConnectionParameters` to also transport an `OCLogMsg` - might be good either way to have that option. Wdyt?

(Very sorry for all the hustle :bow: I will remove `OC_Map` latest by 20.04 and get rid of this duality.)

Reply:
I think it works slightly similar in `OC_Map`, if you look at `get_cluster` it will raise if the object is of type `OCLogMsg`. So it would fail in this instance

Reply:
Just tried locally: 
```
[2023-03-21 10:01:49] [ERROR] [openshift_base.py:init_specs_to_fetch:153] - [app-sre-stage-01] server URL mismatch
[2023-03-21 10:01:49] [ERROR] [openshift_base.py:init_specs_to_fetch:153] - [app-sre-stage-01] server URL mismatch

Process finished with exit code 1
```

Reply:
Awesome! Thanks for checking! :bow: